### PR TITLE
RAC-575: Add Measurement family domain events + Fix CI configuration

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Introduces Domain events for measurement families
   - MeasurementFamilyCreated
   - MeasurementFamilyUpdated
   - MeasurementFamilyDeleted

Events are created & dispatched by the dedicated application services (ideally, they would have been created by the aggregate and dispatched by the repository, that was too much work).

⚠️ Acceptance & EndToEnd tests were not run by the CI, hence I added them in the phpunit config.

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Migration & Installer
- [x] PM Validation (Story)
- [x] Changelog (maintenance bug fixes)
- [x] Tech Doc

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php [hunk 1]
```diff
@@ -21,22 +23,30 @@ class CreateMeasurementFamilyHandler
...
         $measurementFamily = MeasurementFamily::create(
-            MeasurementFamilyCode::fromString($saveMeasurementFamilyCommand->code),
+            $measurementFamilyCode,
             LabelCollection::fromArray($saveMeasurementFamilyCommand->labels),
             UnitCode::fromString($saveMeasurementFamilyCommand->standardUnitCode),
             $units
         );
 
         $this->measurementFamilyRepository->save($measurementFamily);
+        $this->eventDispatcher->dispatch(new MeasurementFamilyCreated($measurementFamilyCode));
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php:49]: Do not call dispatch function if eventDispatcher is null (Same for DeleteMeasurementFamily and SaveMeasurementFamily)

```suggestion
        if ($this->eventDispatcher != null) {
            $this->eventDispatcher->dispatch(new MeasurementFamilyCreated($measurementFamilyCode));
        }
```

**Review by juliensnz — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php [hunk 2]
```diff
@@ -76,6 +76,22 @@ public function normalizeWithIndexedUnits(): array
...
+            throw new UnitNotFoundException();
+        }
+
+        return $unit->getLabel($localeIdentifier);
+    }
+
+    public function getRecordedEvents(): array
+    {
+        return $this->recordedEvents;
+    }
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php:93]: You can remove this function you didn't use this

File: src/Akeneo/Tool/Bundle/MeasureBundle/Validation/Common/LabelCollectionValidator.php [hunk 3]
```diff
@@ -52,7 +52,7 @@ private function validateLocaleCode(ValidatorInterface $validator, $localeCode):
         }
     }
 
-    private function validateLabelForLocale(ValidatorInterface $validator, string $label, string $localeCode): void
+    private function validateLabelForLocale(ValidatorInterface $validator, $label, $localeCode): void
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Validation/Common/LabelCollectionValidator.php:55]: There is a reason for this change ? It seems it's a mistake

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        $this->isThereAtLeastOneAttributeConfiguredWithMeasurementFamily->setStub(true);
+
+        $deleteCommand = new DeleteMeasurementFamilyCommand();
+        $deleteCommand->code = $measurementFamilyCode;
+
+        $violations = $this->validator->validate($deleteCommand);
+        $this->deleteMeasurementFamilyHandler->handle($deleteCommand);
+
+        $this->assertCannotRemoveTheMeasurementFamily($violations);
+        $this->assertMeasurementFamilyEventDispatched($measurementFamilyCode);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php:77]: This event should not be dispatched

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/SaveMeasurementFamilyTest.php [hunk 5]
```diff
@@ -19,17 +20,10 @@
...
-
-    /** @var InMemoryMeasurementFamilyRepository */
-    private $measurementFamilyRepository;
-
-    /** @var SaveMeasurementFamilyHandler */
-    private $saveMeasurementFamilyHandler;
-
-    /** @var InMemoryIsThereAtLeastOneAttributeConfiguredWithMeasurementFamilyStub */
-    private $isThereAtLeastOneAttributeConfiguredWithMeasurementFamily;
+    private ValidatorInterface$validator;
```
Comment (StevenVAIDIE) [inline src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/SaveMeasurementFamilyTest.php:23]:
```suggestion
    private ValidatorInterface $validator;
```

File: src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php (hunk 2 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Model/MeasurementFamily.php:93]: yes done!

File: src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php (hunk 1 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php:49]: yes you are right! done!

File: src/Akeneo/Tool/Bundle/MeasureBundle/Validation/Common/LabelCollectionValidator.php (hunk 3 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Validation/Common/LabelCollectionValidator.php:55]: If the label is an integer, we end up with a php error on the type of the parameter when we we validate in this function that the label is a string to give a nice message to the user :/.
I know it's not perfect but it's better than having a 500 error

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php (hunk 4 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php:77]: I didn't say which event, i said an event is dispatched ^^ let me make this clearer :) $this->assertMeasurementFamilyDeletedEventDispatched :)

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/SaveMeasurementFamilyTest.php (hunk 5 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/SaveMeasurementFamilyTest.php:23]: fixed

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php [hunk 6]
```diff
@@ -4,13 +4,15 @@
...
 
+use Akeneo\Tool\Bundle\MeasureBundle\Event\MeasurementFamilyCreated;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\LabelCollection;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\MeasurementFamily;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\MeasurementFamilyCode;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\Operation;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\Unit;
 use Akeneo\Tool\Bundle\MeasureBundle\Model\UnitCode;
 use Akeneo\Tool\Bundle\MeasureBundle\Persistence\MeasurementFamilyRepositoryInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php:15]: Is there a reason to not use Symfony's interface? (I know it uses PSR under the hood, just askinnnnnnn)
```suggestion
use Symfony\Component\EventDispatcher\EventDispatcherInterface;
```

**Review by ValentinMumble — APPROVED** [review]
GTMMMMMMMMM

File: src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php (hunk 4 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/tests/Acceptance/DeleteMeasurementFamilyTest.php]: YOU ARE RIGHT! seen IRL

File: src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php (hunk 6 above)
Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Application/CreateMeasurementFamily/CreateMeasurementFamilyHandler.php:15]: I think it's equivalent, Psr is more "general" than symfony (i'm good either way)

**Pull request merged**